src/gui/spatialArrow.js

```diff
@@ -0,0 +1,324 @@
+                screenY = screenXY.y;
+                
+                if (lastScreenX !== undefined && lastScreenY !== undefined) {
+                    if ((lastScreenX > 0 && lastScreenY < 0 && screenX < 0 && screenY > 0) || (lastScreenX < 0 && lastScreenY > 0 && screenX > 0 && screenY < 0)) {
+                        // console.warn('reached singularity point!!!');
+                        flag = true;
+                    } else {
+                        flag = false;
+                    }
+                }
+
+                // console.log(screenX, screenY, lastScreenX, lastScreenY);
+                
+                if (flag === true) {
+                    // todo: figure out why this method won't work
+                    // todo: replicate the problem: uncomment below code -> run remote operator -> make cylinder indicators go out of left bound -> make cylinder indicators go out of upper bound -> the arrows are on the left bound instead of upper bound
+                    // todo: replicate the problem: it seems that from left -> upper / upper -> left transitions always have issues.
+                    // todo: other transitions don't have any issue.
+                    // todo: take a look at my screen inequality sketch, figure out why transition between top & left side cause problem
+                    // screenX *= -1;
+                    // screenY *= -1;
+                }
+
+                // todo: from unity tutorial. Figure out why this won't work
+                // // 相当于 "float angle = Mathf.Atan2(screenPos.y, screenPos.x) 里的 screenPos.y, screenPos.x
+                // desX = Remap(screenX, 0, screenW, -screenW/2, screenW/2);
+                // desY = Remap(screenY, 0, screenH, -screenH/2, screenH/2);
+                //
+                // angle = Math.atan2(desY, desX);
+                // angle -= Math.PI / 2;
+                //
+                // let cos = Math.cos(angle);
+                // let sin = -Math.sin(angle);
+                // let m = cos / sin;
+                //
+                // // 相当于两个 screen bounds
+                // desX *= screenBorderFactor;
+                // desY *= screenBorderFactor;
+                //
+                // // check up and down first
+                // if (cos > 0) {
+                //     // up
+                //     finalPosX = desY / m;
+                //     finalPosY = desY;
+                // } else {
+                //     // down
+                //     finalPosX = -desY / m;
+                //     finalPosY = -desY;
+                // }
+                // // out of bounds
+                // if (screenX > desX) {
+                //     finalPosX = desX;
+                //     finalPosY = desX * m;
+                // } else if (screenX < desX) {
+                //     finalPosX = -desX;
+                //     finalPosY = -desX * m;
+                // }
+                // Arrow(finalPosX, finalPosY, angle, 5);
```

Comment:
Can remove this commented code too

Reply:
Yes. Just removed all the unused code in spatialArrow.js